ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/momentumBasedController/feedbackController/taskspace/PointFeedbackController.java

```diff
@@ -610,34 +610,39 @@ private void computeIntegralTerm(FrameVector3D feedbackTermToPack)
          feedbackTermToPack.setToZero(controlFrame);
          yoErrorPositionIntegrated.setToZero(trajectoryFrame);
          yoErrorPositionIntegrated.setCommandId(currentCommandId);
-         return;
       }
-
-      // If the trajectory frame changed reset the integration.
-      if (yoErrorPositionIntegrated.getReferenceFrame() != trajectoryFrame)
+      else
       {
-         yoErrorPositionIntegrated.setToZero(trajectoryFrame);
-      }
-
-      feedbackTermToPack.setIncludingFrame(yoErrorPosition);
-      feedbackTermToPack.scale(dt);
-      feedbackTermToPack.add(yoErrorPositionIntegrated);
-      feedbackTermToPack.changeFrame(controlFrame);
-      selectionMatrix.applyLinearSelection(feedbackTermToPack);
-      feedbackTermToPack.clipToMaxNorm(maximumIntegralError);
-      yoErrorPositionIntegrated.setIncludingFrame(feedbackTermToPack);
-      yoErrorPositionIntegrated.changeFrame(trajectoryFrame);
-      yoErrorPositionIntegrated.setCommandId(currentCommandId);
+         // If the trajectory frame changed reset the integration.
+         if (yoErrorPositionIntegrated.getReferenceFrame() != trajectoryFrame)
+         {
+            yoErrorPositionIntegrated.setToZero(trajectoryFrame);
+         }
+         if (yoErrorPosition.getReferenceFrame() != trajectoryFrame)
```

Comment:
Do you think its worth adding a comment saying that "if the trajectory frame changed, reset the position, or maybe adding to the other comment"

Reply:
Oh. it sounds good. I'll add that comments into the code

Reply:
I have the same comment here as before.